src/test/mockOutputChannel.ts

```diff
@@ -0,0 +1,47 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+
+export class MockOutputChannel implements vscode.OutputChannel {
+  public value: string = ''
+  public isHidden: boolean = false
+  public preserveFocus: boolean = false
+
+  public readonly name = 'Mock channel'
+
+  public append(value: string): void {
+    this.value += value
+  }
+
+  public appendLine(value: string) {
+    this.value += value + '\n'
+  }
+
+  public clear(): void {
+    this.value = ''
+  }
+
+  public dispose(): void {
+    this.value = ''
+  }
+
+  public hide(): void {
+    this.isHidden = true
+  }
+
+  /**
+   * This is overloaded by VS Code but viewColumn version is deprecated thus
+   * show(preserveFocus?: boolean) is really what we should consider.
+   */
+  public show(columnOrPreserveFocus?: vscode.ViewColumn | boolean, preserveFocus?: boolean): void {
```

Comment:
The implementation is good--for consistency, please add the overload signatures above it:

```typescript
public show(preserveFocus?: boolean): void
public show(column?: ViewColumn, preserveFocus?: boolean): void
public show(columnOrPreserveFocus?: vscode.ViewColumn | boolean, preserveFocus?: boolean): void {
    // ...
}
```

Reply:
That exists in the OutputChannel interface defined by VS Code, which `MockOutputChannel` formally implements.

https://code.visualstudio.com/api/references/vscode-api#OutputChannel
https://github.com/Microsoft/vscode/blob/7c1fe9a7d6eaa6b3bbfddb740996c4bb867e6385/src/vs/vscode.d.ts